simapp/abci.go

```diff
@@ -75,10 +185,63 @@ func (h *VoteExtensionHandler) VerifyVoteExtension() sdk.VerifyVoteExtensionHand
 		case !bytes.Equal(req.Hash, ve.Hash):
 			return &abci.ResponseVerifyVoteExtension{Status: abci.ResponseVerifyVoteExtension_REJECT}, nil
 
-		case len(ve.Data) != 1024:
-			return &abci.ResponseVerifyVoteExtension{Status: abci.ResponseVerifyVoteExtension_REJECT}, nil
 		}
-
 		return &abci.ResponseVerifyVoteExtension{Status: abci.ResponseVerifyVoteExtension_ACCEPT}, nil
 	}
 }
+
+// PreFinalizeBlock -> Extract VEs from previous block and persist in DB.
+func (app *SimApp) NewPreFinalizeBlockHookHandler() sdk.PreFinalizeBlockHook {
+	return func(ctx sdk.Context, req *abci.RequestFinalizeBlock) error {
+		if req.Height > 5 {
+			var ve VoteExtension
+			var eventData *stakingtypes.EventRecord
+			for _, v := range req.Txs {
+				// Have a way to segregate b/w signed VEs and regular txs. Assuming for VEs, byte length would be 8
+				if len(v) >= 8 {
```

Comment:
Is this always the case? Can other txs have the same length?

Reply:
The point here is that we will need a distinct way of having separation between normal txs and V.Es. That's why the comment :
`// Have a way to segregate b/w signed VEs and regular txs. Assuming for VEs, byte length would be 8
`